src/utils/transaction.ts

```diff
@@ -292,58 +292,54 @@ const transaction = {
    * Normal txs can be "unbalanced" when minting or melting tokens, but since we are not required to add the minted tokens on the inputs
    * Or conversely add the melted tokens on the outputs, we will ignore minted/melted funds.
    *
+   * @param {string} token The token we want to calculate the balance.
    * @param {IDataTx} tx The transaction we want to calculate the balance.
-   * @returns {Promise<Map<string, Record<'funds'|'mint'|'melt', number>>>} The balance of all tokens on the transaction.
+   * @returns {Promise<Record<'funds'|'mint'|'melt', number>>} The balance of the given token on the transaction.
    */
-  async calculateTxBalanceToFillTx(tx: IDataTx): Promise<Map<string, Record<'funds'|'mint'|'melt', number>>> {
-    function getEmptyBalance(): Record<'funds'|'mint'|'melt', number> {
-      return {'funds': 0, 'mint': 0, 'melt': 0};
-    }
-    const balance = new Map<string, Record<'funds'|'mint'|'melt', number>>();
-
+  async calculateTxBalanceToFillTx(token: string, tx: IDataTx): Promise<Record<'funds'|'mint'|'melt', number>> {
```

Comment:
Not really, but it came from a breakdown of the `fillTx` method which is async so by default i made it async.
After thinking about this method, maybe we should include it back into `fillTx` since it can only be used there.
But I think this should be made on its own PR.